docs/design/ocs-to-ocs-api.md

```diff
@@ -0,0 +1,96 @@
+# FEATURE: OCS to OCS API changes
+
+## Summary
+Allow customers to share OCS storage across multiple OCP clusters. Each
+consumer OCP cluster will have OCS deployed that connects to the provider
+cluster.
+
+## Motivation
+This feature gives flexibility to customers and lets them share storage and
+better utilize their OpenShift clusters.
+
+### Goals
+Allow a customer to have a single OCS cluster sharing the storage across
+multiple OCP clusters. Each consumer OCP cluster will have OCS deployed in
+external mode that connects to an internal mode provider cluster. Also, allow
+us to reduce the cost of storage and maintenance.
+
+### Non-Goals
+Networking b/w two OCP clusters.
+
+## Proposal
+Introduce additional StorageCluster spec fields to support a new external mode
+where the storage is an OCS cluster.
+
+## Design Details
+
+### API changes required for a provider deployment
+```yaml
+spec:
+    allowRemoteStorageConsumers: <bool>
+```
+
+* `spec.allowRemoteStorageConsumers` Indicates that the OCS cluster should
+deploy the needed components to enable connections from remote consumers.
+When enabled the operator will deploy an API server that will allow onboarding,
+offboarding and second-day operation requests coming from consumer clusters.
+
+- onboarding: Configure the consumer to use the storage provided by the
+provider cluster. The consumer will install the OCS in external mode and
+connect to the provider cluster.
+- offboarding: Uninstall the ocs cluster and disconnect it from the provider.
+
+### API changes required for a consumer deployment
+```yaml
+spec:
+    externalStorage:
+        enable: <bool> (existing field)
+        storageProviderKind: <string> (ocs/rhcs) (default:rhcs)
+        connectionString: <string>
+        requestedCapacity: <resource.Quantity> (default:1Ti)
+status:
+    externalStorage:
+        actualCapacity: <resource.Quantity>
+```
+
+* `spec.externalStorage.storageProviderKind` Identify the type of storage
+provider cluster this consumer cluster is going to connect to.
+  - A value of RHCS will indicate that the consumer cluster is going to connect
+  to and consume storage from a **rhcs** deployment (the legacy external mode).
+  - A value of **ocs-provider** will indicate that the consumer cluster is
+  going to connect to and consume storage from an ocs storage provider cluster.
+* `spec.externalStorage.connectionString` An encoded string holding connection
+and identity information that is needed in order to establish connection with
+the storage providing cluster.
+* `spec.externalStorage.requestedCapacity` Will define the desired capacity
+requested by a consumer cluster deployment.
+* `status.externalStorage.actualCapacity` Will report the actual capacity
+granted to the consumer cluster by the provider cluster.
+
+#### There are be 3 types of operations which will be added to the ocs-operator
+* **OnBoarding:** ocs-operator, running on a consumer cluster, will reach out
```

Comment:
Based on the `Revised Consumer Onboarding Proposal`,  OnBoarding api should also create the StorageConsumer CR in the provider cluster.